src/features/environmentCreate/components/Specification/SpecificationCreate.tsx

```diff
@@ -59,21 +66,34 @@ export const SpecificationCreate = ({ onCreateEnvironment }: any) => {
       dispatch(
         editorCodeUpdated({
           channels: editorContent.channels,
-          dependencies: editorContent.dependencies
+          dependencies: editorContent.dependencies,
+          variables: editorContent.variables
         })
       );
     } else {
-      setEditorContent({ dependencies: requestedPackages, channels });
+      setEditorContent({
+        dependencies: requestedPackages,
+        variables,
+        channels
+      });
     }
 
     setShow(value);
   };
 
-  const formatCode = (channels: string[], dependencies: string[]) => {
-    if (channels.length === 0 && dependencies.length === 0) {
-      return "channels:\n  -\ndependencies:\n  -";
+  const formatCode = (
+    channels: string[],
+    dependencies: string[],
+    variables: Record<string, string>
+  ) => {
+    if (
+      channels.length === 0 &&
+      dependencies.length === 0 &&
+      Object.keys(variables).length === 0
+    ) {
+      return "channels:\n  -\ndependencies:\n  -\nvariables: {}";
     }
-    return stringify({ channels, dependencies: dependencies });
+    return stringify({ channels, dependencies, variables });
```

Comment:
How the custom pretty-printer looks when no data is there:

<img width="208" alt="Screen Shot 2024-01-08 at 03 43 00" src="https://github.com/conda-incubator/conda-store-ui/assets/10469885/881be8fb-dcc2-4235-a228-8d413efcec17">